components/engine_rocks/src/table_properties.rs

```diff
@@ -1,9 +1,44 @@
 // Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
 
-use crate::engine::RocksEngine;
-use crate::util;
-use engine_traits::Range;
-use engine_traits::{Error, Result};
+use crate::{util, RocksEngine};
+use engine_traits::{Error, Range, Result};
+
+#[repr(transparent)]
+pub struct UserCollectedProperties(rocksdb::UserCollectedProperties);
+impl engine_traits::UserCollectedProperties for UserCollectedProperties {
+    fn get(&self, index: &[u8]) -> Option<&[u8]> {
+        self.0.get(index)
+    }
+}
+
+#[repr(transparent)]
+pub struct TablePropertiesCollection(rocksdb::TablePropertiesCollection);
+impl engine_traits::TablePropertiesCollection for TablePropertiesCollection {
+    type UserCollectedProperties = UserCollectedProperties;
+    fn iter_user_collected_properties<F>(&self, mut f: F)
```

Comment:
can we implement `Iterator` for it?

Reply:
For the rocksdb implementation it's hard because `Iterator` is implemented for `rocksdb::TablePropertiesCollectionIter` instead of `rocksdb::TablePropertiesCollection`. So exposing an `Iterator` will require another `struct CollectionIter`. So here I use a closure to avoid too many structs.